src/hooks/__tests__/useData.tests.ts

```diff
@@ -0,0 +1,132 @@
+    genres: ['Criminal', 'Romance'],
+    runtime: 200,
+  },
+];
+
+describe('useData', () => {
+  beforeEach(() => {
+    fetchMock.resetMocks();
+    fetchMock.mockResponseOnce(JSON.stringify({ data: movies }));
+  });
+
+  it('should return fetch data response', async () => {
+    const { result } = await renderHook(() => useData('', '', 'Title'));
+
+    expect(fetchMock).toHaveBeenCalledTimes(1);
+    await waitFor(() => {
+      expect(result.current).toEqual(movies);
+    });
+  });
+
+  it('should not contain filter if "all" passed', () => {
+    renderHook(() => useData('', 'All', 'Title'));
+
+    expect(fetchMock).toHaveBeenCalledWith(
+      expect.not.stringContaining('filter'),
+      expect.anything()
+    );
+  });
+
+  it('should contain filter param with genre if genre passed', () => {
+    const genre = 'Comedy';
+    renderHook(() => useData('', genre, 'Title'));
+
+    expect(fetchMock).toHaveBeenCalledWith(
+      expect.stringContaining('filter'),
+      expect.anything()
+    );
+    expect(fetchMock).toHaveBeenCalledWith(
+      expect.stringContaining(genre),
+      expect.anything()
+    );
+  });
+
+  it('should not contain search param if search query not passed', () => {
+    renderHook(() => useData('', '', 'Title'));
+
+    expect(fetchMock).toHaveBeenCalledWith(
+      expect.not.stringContaining('search'),
+      expect.anything()
+    );
+    expect(fetchMock).toHaveBeenCalledWith(
+      expect.not.stringContaining('searchBy'),
+      expect.anything()
+    );
+  });
+
+  it('should contain search param if search query passed', () => {
+    const searchQuery = 'The';
+    renderHook(() => useData(searchQuery, '', 'Title'));
+
+    expect(fetchMock).toHaveBeenCalledWith(
+      expect.stringContaining('search'),
+      expect.anything()
+    );
+    expect(fetchMock).toHaveBeenCalledWith(
+      expect.stringContaining(searchQuery),
+      expect.anything()
+    );
+    expect(fetchMock).toHaveBeenCalledWith(
+      expect.stringContaining('searchBy=title'),
+      expect.anything()
+    );
+  });
+
+  it('should contain sort by title param if sort title passed', () => {
+    renderHook(() => useData('', '', 'Title'));
+
+    expect(fetchMock).toHaveBeenCalledWith(
+      expect.stringContaining('sortBy=title&sortOrder=asc'),
+      expect.anything()
+    );
+  });
+
+  it('should contain sort by release_date param if sort by release date passed', () => {
+    renderHook(() => useData('', '', 'Release Date'));
+
+    expect(fetchMock).toHaveBeenCalledWith(
+      expect.stringContaining('sortBy=release_date&sortOrder=asc'),
+      expect.anything()
+    );
+  });
+
+  it('should return empty array if rejected', async () => {
```

Comment:
It will return old data if the next request is rejected. Is it expected?

Reply:
Fixed in hook